mithril-stm/src/multi_sig.rs

```diff
@@ -589,6 +535,95 @@ impl_serde!(VerificationKey, VerificationKeyVisitor, 96);
 impl_serde!(ProofOfPossession, ProofOfPossessionVisitor, 96);
 impl_serde!(Signature, SignatureVisitor, 48);
 
+// ---------------------------------------------------------------------
+// Unsafe helpers
+// ---------------------------------------------------------------------
+mod unsafe_helpers {
+    use super::*;
+    use blst::{
+        blst_fp12, blst_fp12_finalverify, blst_p1_affine_generator, blst_p2_affine_generator,
+        blst_sk_to_pk_in_g1,
+    };
+
+    pub(crate) fn verify_miller_loop(vk: &BlstVk, k2: &blst_p1) -> bool {
+        unsafe {
+            let g1_p = *blst_p1_affine_generator();
+            let mvk_p = std::mem::transmute::<&BlstVk, &blst_p2_affine>(vk);
+            let ml_lhs = blst_fp12::miller_loop(mvk_p, &g1_p);
```

Comment:
As mentioned above, let's try, as much as we can, to use the mithril-stm types in the function signatures, rather than the blst types. This allows us to be more consistent in the usage of transmute. i.e., we always transmute by value rather than by reference:
```suggestion
    pub(crate) fn verify_pairing(vk: & VerificationKey, pop: & ProofOfPossession) -> bool {
        unsafe {
            let g1_p = *blst_p1_affine_generator();
            let mvk_p = std::mem::transmute::<BlstVk, blst_p2_affine>(vk.0);
            let ml_lhs = blst_fp12::miller_loop(&mvk_p, &g1_p);
```